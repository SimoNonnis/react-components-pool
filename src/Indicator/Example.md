Indicator renders a green circle if status prop is "ready"

```jsx
<Indicator status="ready" />
```

Indicator renders an orange circle if status prop is "awaiting"

```jsx
<Indicator status="awaiting" />
```
