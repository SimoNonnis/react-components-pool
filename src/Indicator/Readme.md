Indicator renders a green circle if status prop is "ready" or an orange one if is "awaiting".
Default status will be null with no background color.

```jsx
<Indicator status="ready" />
```

```jsx
<Indicator status="awaiting" />
```
