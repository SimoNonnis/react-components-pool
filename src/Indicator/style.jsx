import styled, { keyframes } from 'styled-components';

const pulse = keyframes`
  30% {
    transform: scale(.7);
  }
  75% {
    transform: scale(1.1);
  }
`;

const getColor = (theme, status) => {
  if (status === 'ready') return theme.color.green;
  if (status === 'awaiting') return theme.color.orange;
  return 'transparent';
};

export const Circle = styled.div`
  width: 7px;
  height: 7px;
  border-radius: 50%;
  background: ${({ theme, status }) => getColor(theme, status)};
  }};
  animation: ${({ status }) =>
    status === 'awaiting' ? `${pulse} 1.3s infinite backwards ease-out` : null};
`;
Circle.displayName = 'Circle';
