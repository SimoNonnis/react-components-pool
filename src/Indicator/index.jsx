import React from 'react';
import PropTypes from 'prop-types';

import { Circle } from './style';

const propTypes = {
  status: PropTypes.oneOf(['awaiting', 'ready']),
};
const defaultProps = {
  status: null,
};

function Indicator({ status }) {
  return <Circle status={status} />;
}

Indicator.propTypes = propTypes;
Indicator.defaultProps = defaultProps;

export default Indicator;
