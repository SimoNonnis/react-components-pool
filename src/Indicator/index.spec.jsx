/* eslint-env jest */
import React from 'react';

import Indicator from './';

describe('Test <Indicator />', () => {
  it('should render Indicator but will not be visible', () => {
    const wrapper = shallow(<Indicator />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Indicator with a green circle if status prop is "ready"', () => {
    const wrapper = shallow(<Indicator status="ready" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Indicator with a orange circle if status prop is "awaiting"', () => {
    const wrapper = shallow(<Indicator status="awaiting" />);
    expect(wrapper).toMatchSnapshot();
  });
});
