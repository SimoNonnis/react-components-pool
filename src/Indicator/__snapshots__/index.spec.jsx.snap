// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Indicator /> should render Indicator but will not be visible 1`] = `
<Circle
  status={null}
/>
`;

exports[`Test <Indicator /> should render Indicator with a green circle if status prop is "ready" 1`] = `
<Circle
  status="ready"
/>
`;

exports[`Test <Indicator /> should render Indicator with a orange circle if status prop is "awaiting" 1`] = `
<Circle
  status="awaiting"
/>
`;
